packages/clerk-js/src/ui/primitives/gapPropertyCompat.ts

```diff
@@ -1,11 +1,11 @@
-export const createFlexGapPropertyIosCompat = (val: string) => {
+export const createFlexGapPropertyIosCompat = (val: string, dir: 'col' | 'row') => {
   // Resolve gap property compatibility issue on iOS 13
   // This is not needed for css grid, only for flex
-  // TODO: Test whether we need to make the check more generic
-  if (navigator?.userAgent?.match(/(iphone|ipad).+os.+13.+safari/i)) {
+  if (navigator?.userAgent?.match(/(iphone|ipad).+(os).*(\s13).+safari/i)) {
     return {
-      '& > * + *': {
-        marginTop: val,
+      '& > *:not([hidden]):not([style*="visibility: hidden"]) + *:not([hidden]):not([style*="visibility: hidden"])': {
+        marginLeft: dir === 'row' ? val : undefined,
+        marginTop: dir === 'col' ? val : undefined,
       },
```

Comment:
This update was to improve the layout rendering for the fallback:

| BEFORE | AFTER |
|--------|--------|
| <img width="612" alt="Screenshot 2024-08-27 at 5 10 36 PM" src="https://github.com/user-attachments/assets/a9f7a4f7-0899-44bb-9085-a7ebbd798b8b"> | <img width="612" alt="Screenshot 2024-08-27 at 5 11 21 PM" src="https://github.com/user-attachments/assets/3f2a8977-2ce9-4739-b3da-235a87c14536"> |